chrome-extension/service.js

```diff
@@ -28,10 +31,39 @@ async function getSentimentAnaylsisMsg(linkUrl) {
 /**  
  * @param {String} text, to be analyzed
  * @returns {String} toxicity analysis message
+ * Modified based on src/main/webapp/scripts/perspective.js
+ * and src/main/webapp/scripts/perspective-api-service.js
  */
 async function getToxicityAnaylsisMsg(text) {
-  // TODO add service later
-  return 'Toxicity score is 24';
+  const request = new Request(
+    `https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=${API_KEY}`,
+    {
+      method: 'POST',
+      body: makePerspectiveRequestString(text)
+    }
+  );
+  return fetch(request)
+    .then(response => response.json())
+    .then((responseJson) => {
+      return toxicityToPercentString(responseJson.attributeScores.TOXICITY.summaryScore.value) + ' toxic';
+    })
+    .catch((err) => {
+      console.error('An error occurred with the Perspective API', err);
+      return 'The analysis feature is unavailable at this moment.';
+    });
+}
+
+function makePerspectiveRequestString(text) {
+  const bodyObject = {
+    'comment': { 'text': text },
+    'requestedAttributes': { 'TOXICITY': {} },
+    'languages': ['en']
+  };
+  return JSON.stringify(bodyObject);
+}
+
+function toxicityToPercentString(toxicity) {
+  return (toxicity * 100).toFixed(2).toString() + '%';
 }
```

Comment:
I updated this function somewhere, can you grab the updated version (toFixed(1) instead and also added some explanation after % to indicate what the % stands for)

Reply:
Done